client/sub_client.c

```diff
@@ -122,7 +122,7 @@ static void my_connect_callback(struct mosquitto *mosq, void *obj, int result, i
 
 	connack_result = result;
 	if(!result){
-		mosquitto_subscribe_multiple(mosq, NULL, cfg.topic_count, cfg.topics, cfg.qos, cfg.sub_opts, cfg.subscribe_props);
+		mosquitto_subscribe_multiple(mosq, NULL, cfg.topic_count, (const char* const*)cfg.topics, cfg.qos, cfg.sub_opts, cfg.subscribe_props);
```

Comment:
Unfortunately C (unlike C++) also warns about converions from `char**` to `const char* const*`, even though the latter has lower requirements.